client/app/scripts/superdesk/editor/editor.js

```diff
@@ -23,6 +23,14 @@ function EditorService() {
 
     var vm = this;
 
+    this.historyBodyHtml = [];
+    this.historyHeadline = [];
+    this.historyAbstract = [];
+
+    this.index = -1;
+    this.indexHeadline = -1;
+    this.indexAbstract = -1;
```

Comment:
It probably makes sense to use an alias for `this` (i.e. `vm`) if we already defined it IMO.